elastic/apm-server
# Move APM-specific template defaults to idxmgmt

## Motivation/summary

Move APM-specific `setup.template.settings` defaults
away from using config overrides, and add the logic
to the idxmgmt package.

Similarly drop `logging.files.rotateeverbytes`, as
we were setting what is already the default defined
by libbeat.

Now when you run `apm-server export config`, these
defaults do not show up at all. The relevant system
tests have been migrated to Go, and the expectations
updated to match the change in behaviour.

We customise `logging.metrics.enabled`, and there
is no other way to change the default before it is
interpreted by libbeat, so it remains in config
overrides for now.

## Checklist


I have considered changes for:
~- [ ] documentation~
~- [ ] logging (add log lines, choose appropriate log selector, etc.)~
~- [ ] metrics and monitoring (create issue for Kibana team to add metrics to visualizations, e.g. [Kibana#44001](https://github.com/elastic/kibana/issues/44001))~
~- [ ] telemetry~
~- [ ] Elasticsearch Service (https://cloud.elastic.co)~
~- [ ] Elastic Cloud Enterprise (https://www.elastic.co/products/ece)~
~- [ ] Elastic Cloud on Kubernetes (https://www.elastic.co/elastic-cloud-kubernetes)~

## How to test these changes

- `apm-server export config` should not show any template config
- Run `apm-server` and check that it sets up templates as expected, including setting the maximum number of fields to 2000.

## Related issues

Pulled out of https://github.com/elastic/apm-server/pull/4449

This is necessary before we can properly warn when data streams and template setup are both configured.
